src/Symfony/Bridge/Doctrine/Form/Type/EntityType.php

```diff
@@ -75,8 +76,23 @@ public function getName()
     public function getQueryBuilderPartsForCachingHash($queryBuilder)
     {
         return array(
-                $queryBuilder->getQuery()->getSQL(),
-                $queryBuilder->getParameters()->toArray(),
+            $queryBuilder->getQuery()->getSQL(),
+            array_map(array($this, 'parameterToArray'), $queryBuilder->getParameters()->toArray()),
         );
     }
+
+    /**
+     * Converts a query parameter to an array.
+     *
+     * @param Parameter $parameter The query parameter
+     *
+     * @return array The array representation of the parameter
+     *
+     * @internal This method is public to be usable as callback. It should not
+     *           be used in user code.
+     */
+    public function parameterToArray(Parameter $parameter)
```

Comment:
You're right, it's only required when accessing methods in other classes: https://3v4l.org/qUi6u